addons/purchase/models/purchase.py

```diff
@@ -984,7 +984,7 @@ def _compute_qty_invoiced(self):
             # compute qty_invoiced
             qty = 0.0
             for inv_line in line._get_invoice_lines():
-                if inv_line.move_id.state not in ['cancel']:
+                if inv_line.move_id.state not in ['cancel'] or inv_line.move_id.payment_state == 'invoicing_legacy':
```

Comment:
The correct way of checking would be to use `payment_state_before_switch` (beware that it is defined in `account_accountant`, which is not necessary for `purchase` or `sale`

A helper function `def is_cancelled(self)` on `account.move`, defined in `account` and  extended in `account_accountant` should make the implementation easy.

Reply:
Hi @william-andre, thanks for your feedbacks.

Before making the changes you suggested, could you help me clarify a few things?

My understanding of this issue is the following: we want SOs and POS to take into account invoices that were posted before an invoicing threshold. That means we need information about the **state** of the invoice before the switch. We don't really care about the payment state, so I'm struggling to understand how `payment_state_before_switch` is useful in our case.

Now, why I'm using `payment_state` is we don't care about it? From what I understand, changing the invoicing threshold has, among others, two effects on **posted** invoices whose invoice date is before the new threshold:
- the invoice's state is set to `cancel`
- and the invoice's payment state is set to `invoicing_legacy`

This means that checking if `payment_state = invoicing_legacy` is the only way we can know if a now canceled invoice was posted before.

Am I missing something?

Thanks!

Reply:
You are right
I didn't see that moves that were already cancelled are not set to `invoicing_legacy`